src/main/java/io/cdap/plugin/successfactors/source/metadata/SuccessFactorsSchemaGenerator.java

```
@@ -0,0 +1,770 @@
+   * @return {@code Schema.Field}
+   */
+  private Schema.Field buildSchemaField(SuccessFactorsColumnMetadata successFactorsColumnDetail) {
+    if (successFactorsColumnDetail.containsChild()) {
+      List<Schema.Field> outputSchema = successFactorsColumnDetail.getChildList()
+        .stream()
+        .map(this::buildSchemaField)
+        .collect(Collectors.toList());
+
+      String typeName = buildNestedTypeUniqueName(successFactorsColumnDetail.getName());
+
+      if (successFactorsColumnDetail.getType().equals(EdmTypeKind.ENTITY.name()) &&
+        (successFactorsColumnDetail.getMultiplicityOrdinal() != null &&
+          successFactorsColumnDetail.getMultiplicityOrdinal() > 0)) {
+        // adding 1 to * multiplicity record to ARRAY type
+        return Schema.Field.of(successFactorsColumnDetail.getName(),
+                               Schema.arrayOf(Schema.recordOf(typeName, outputSchema)));
+      }
+
+      // adding 0 to 1 multiplicity record to NULLABLE
+      return Schema.Field.of(successFactorsColumnDetail.getName(),
+                             Schema.nullableOf(Schema.recordOf(typeName, outputSchema)));
+    }
+
+    return Schema.Field.of(successFactorsColumnDetail.getName(), buildRequiredSchemaType(successFactorsColumnDetail));
+  }
+
+  /**
+   * Build and returns the appropriate schema type.
+   *
+   * @param successFactorsColumnDetail {@code SuccessFactorsColumnMetadata}
+   * @return {@code Schema}
+   */
+  private Schema buildRequiredSchemaType(SuccessFactorsColumnMetadata successFactorsColumnDetail) {
+    Schema schemaType = SCHEMA_TYPE_MAPPING.get(successFactorsColumnDetail.getType());
+
+    if (schemaType.getLogicalType() == Schema.LogicalType.DECIMAL
+      && successFactorsColumnDetail.getPrecision() != null
+      && successFactorsColumnDetail.getScale() != null) {
+
+      schemaType = Schema.decimalOf(successFactorsColumnDetail.getPrecision(), successFactorsColumnDetail.getScale());
+    }
```

If precision and scale are not specified in successFactorsColumnDetail, then we default to (2, 1)? Can that happen? Should we throw an exception if precision/scale are not available?

---

Yes, we default to (2,1). And yes, this can happen. We saw a lot of records where no value was specified for precision and scale. Throwing an exception, if precision/scale is not available, will cause the pipeline to fail. So, the `null` check was added and default values were provided. 

As (2,1) may not capture all the decimal places correctly, we had updated the precision to 15 and the scale to 2.

Correcting it here as well. 288b022add0e0a7a19652bd3d4efff4b18f3bf2f